runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -394,6 +394,13 @@ protected long getMaxGlobalFromMaxEntries(int maxEntries) {
         StreamAddressSpace streamAddressSpace = runtime.getSequencerView()
                 .getStreamAddressSpace(new StreamAddressRange(getId(), Address.MAX, getCurrentGlobalPosition()));
 
+        // If sequencer does not return an address map for this stream, it means it is an empty table, we can
+        // safely return because CorfuStore APIs have validated the table in question is present in the RegistryTable
+        // before hitting this API
+        if (streamAddressSpace == null) {
+            return Address.NON_ADDRESS;
+        }
+
```

Comment:
I think this is a bug in `SequencerServer::getStreamsAddressesMap`, it shouldn't return nulls for streams that are empty. It should return an empty `StreamAddressSpace` with the trimMark set to `Address.NON_EXIST`. Then part will take care of the empty table issue:

```java

long size = streamAddressSpace.size();
        if (size == 0L) {
            return Address.NON_ADDRESS;
        }

```

Reply:
I agree, the only concern I have here is wether if this would affect the cases where a stream really does not exist (other code paths), could it? if not I can move it there...

Reply:
Done